File: application-entrainement/backend/app/models/workout.py
from sqlalchemy import Column, Integer, String, DateTime, Boolean, Text, JSON
from datetime import datetime
from ..database import Base

class Workout(Base):
    __tablename__ = "workouts"

    id = Column(Integer, primary_key=True, index=True)
    workout_type = Column(String, index=True)  # Endurance, Tempo, Seuil, VO2 Max, Sprint
    name = Column(String, index=True)
    duration_minutes = Column(Integer)
    difficulty_level = Column(Integer)  # 1 to 10
    description_short = Column(String)
    description_long = Column(Text)
    
    # Training scheme stored as JSON
    # Structure example: [{"type": "interval", "pace": "4:00", "duration": 5, "distance": 1.25, "repetitions": 1}]
    scheme = Column(JSON, nullable=True)
    
    # Execution data
    date = Column(DateTime, index=True)
    is_validated = Column(Boolean, default=False)
    perceived_difficulty = Column(Integer, nullable=True)
    athlete_comment = Column(Text, nullable=True)
    
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

File: application-entrainement/frontend/src/types/workout.ts
export type WorkoutType = 'Endurance' | 'Tempo' | 'Seuil' | 'VO2 Max' | 'Sprint' | 'Libre' | 'Trail' | 'Fractionné' | 'Sortie Longue';

export interface WorkoutInterval {
  type: string;
  pace: string;
  duration?: number;
  distance?: number;
  repetitions: number;
}

export interface Workout {
  id: number;
  workout_type: string;
  category: string;
  name: string;
  duration_minutes: number;
  distance_km: number | null;
  difficulty_level: number;
  estimated_load: number | null;
  description_short: string;
  description_long: string;
  scheme: WorkoutInterval[] | null;
  date: string;
  is_validated: boolean;
  perceived_difficulty: number | null;
  athlete_comment: string | null;
  plan_id: number | null;
  athlete_id: number | null;
  created_at: string;
  updated_at: string;
}

export interface CatalogInterval {
  type: string;
  pace_vma: number; // en %
  duration?: number;
  distance?: number;
  repetitions: number;
}

export interface CatalogWorkout {
  id: number;
  name: string;
  workout_type: string;
  category: string;
  perceived_difficulty: number;
  scheme: CatalogInterval[];
}
